infra/aws/terraform/prow-build-cluster/docs/EKSUpgradeRunbook.md

```diff
@@ -0,0 +1,87 @@
+# EKS Kubernetes Version Upgrade
+
+## Control Plane Upgrade
+
+To upgrade the Kubernetes version, you need to update the `cluster_version` variable on the cluster's `terraform.tfvars` file:
+
+```diff
+## canary cluster:
+#infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
+
+## prod cluster:
+#infra/aws/terraform/prow-build-cluster/terraform.prod.tfvars
+
+--- a/infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
++++ b/infra/aws/terraform/prow-build-cluster/terraform.canary.tfvars
+@@ -31,13 +31,13 @@ eks_cluster_admins = [
+ ]
+ 
+ cluster_name               = "prow-canary-cluster"
+-cluster_version            = "1.28"
++cluster_version            = "1.29"
+```
+
+After this change, you need to run `make plan` and `make apply` commands. (Checkout [IaC Document](./IaC.md) for more information).
+
+
+## Worker Nodes Upgrade
+
+### Karpenter
+
+The current configuraiotn for Karpenter NodePool uses `amiFamily` setting. As a result, any new nodes provisioned after the cluster upgrade will automatically use the latest AMI for the new Kubernetes version.
+
+If you need to upgrade already provisioned nodes immediately, you should manually drain the node(s), Karpenter will automatically remove the node due to emptiness. Here is an example:
+
+Get our Karpenter managed node list:
+```bash
+$> kubectl get nodes -l karpenter.sh/nodepool=default
+NAME                                        STATUS   ROLES    AGE     VERSION
+ip-10-0-125-19.us-east-2.compute.internal   Ready    <none>   4h12m   v1.29.5-eks-1109419
+ip-10-0-164-0.us-east-2.compute.internal    Ready    <none>   4h12m   v1.29.5-eks-1109419
+ip-10-0-4-161.us-east-2.compute.internal    Ready    <none>   4h13m   v1.29.5-eks-1109419
+ip-10-0-53-105.us-east-2.compute.internal   Ready    <none>   4h13m   v1.29.5-eks-1109419
+```
+
+Drain one of the nodes:
```

Comment:
Can we clarify that this shouldn't be done if jobs are running, and in that case you should cordon it first, wait for jobs to be done, and then delete it?

Reply:
Ok, we will just cordon the node instead of draining immediately. There will not be new jobs assigned to the node, and Karpenter will take care of the rest when the node is "empty."